transport/tchannel/handler.go

```diff
@@ -155,6 +150,14 @@ func (h handler) callHandler(ctx context.Context, call inboundCall, start time.T
 
 	spec, err := h.router.Choose(ctx, treq)
 	if err != nil {
+		if _, ok := err.(errors.UnrecognizedProcedureError); ok {
```

Comment:
would you be ok to do an early "return err" here instead? and similar for the sub-conditions